app/controllers/api/v4/runs_controller.rb

```diff
@@ -14,11 +14,30 @@ class Api::V4::RunsController < Api::V4::ApplicationController
   end
 
   def show
-    if params[:historic] == '1'
-      @run.parse(fast: false)
-      render json: @run, serializer: Api::V4::RunWithHistorySerializer
+    accept = request.headers.fetch('HTTP_ACCEPT', 'json').downcase
+    if accept == "original_timer"
+      accept = @run.program
+    end
+    timer = Run.program(accept)
+    if accept == 'json' || timer.nil? || !Run.exportable_programs.include?(timer)
+      if params[:historic] == '1'
+        @run.parse(fast: false)
+        render json: @run, serializer: Api::V4::RunWithHistorySerializer
+      else
+        render json: @run, serializer: Api::V4::RunSerializer
+      end
     else
-      render json: @run, serializer: Api::V4::RunSerializer
+      send_data(
+        if timer == Run.program(@run.timer)
+          @run.file
+        else
+          render_to_string(file: "runs/#{accept}.html.erb", layout: false)
+        end,
+        layout: false,
+        type: "application/#{accept}",
+        disposition: 'inline',
+        filename: "#{@run.filename(timer: timer)}"
+      )
```

Comment:
This `send_data` looks similar to the code we use to make a user's browser initiate a download (rather than rendering text in the browser). Do we know what the technical difference is that causes browsers to treat it differently, and do we know that it won't throw API consumers?

Reply:
From what I was able to find, `send_data` allows us to send arbitrary data down the pipe.  What do you mean by throw api consumers?

Reply:
By throw I mean mess up how they expect their data to arrive. I looked it up and it's a `Content-Disposition` header; if it's `inline`, the browser will render the text, if it's `attachment`, the browser will download the text as a file.

So it looks like there's no actual difference in the body itself, and I can't immediately find anything online saying if/how APIs are expected to set this header, so I don't think we need to worry about anything here.

Reply:
I'll write a quick script to download from the endpoint and make 100% sure that the programs can open them without any errors.